codice/ddf-ui
# Adding Boolean Search Bar to Basic and Advanced Query

Description

Implemented support for Boolean statements in basic and advanced search forms. The Boolean search bar supports AND, OR and NOT operators and will automatically suggest them in correct places as well as validate them. 

The default value is "*" as the quotation marks are required for special character handling in the backend search parser (this requirement may be changed)
<img width="310" alt="Screen Shot 2021-06-16 at 11 36 55 AM" src="https://user-images.githubusercontent.com/22224756/122279458-e8bf4000-ce9c-11eb-9381-e4c1552a9958.png">

Below is the auto suggest of the boolean operators
<img width="310" alt="Screen Shot 2021-06-16 at 11 37 08 AM" src="https://user-images.githubusercontent.com/22224756/122279504-f5439880-ce9c-11eb-8558-a55f1dbe64ca.png">

If the boolean statement is not completed the query will be invalid and the below message will show
<img width="310" alt="Screen Shot 2021-06-16 at 11 37 31 AM" src="https://user-images.githubusercontent.com/22224756/122279519-fbd21000-ce9c-11eb-9055-be1d1d036631.png">

Below shows how the NOT operator behaves. Selecting it will put the operator in the textbox with a set of parenthesis around the term you wish to use the NOT on. Failing to put something in the parenthesis will result in an error.
<img width="310" alt="Screen Shot 2021-06-16 at 11 37 52 AM" src="https://user-images.githubusercontent.com/22224756/122281982-bcf18980-ce9f-11eb-8256-5e2107cc457d.png">

Below shows the Boolean Search Bar within the advanced search. It behaves the same as in the basic search and the text in the basic form will carry over when switching to advanced. If the boolean statement is not correct, it will be ignored when the search is executed and results will be returned for the other search attributes.
<img width="310" alt="Screen Shot 2021-06-16 at 11 38 09 AM" src="https://user-images.githubusercontent.com/22224756/122279569-0a202c00-ce9d-11eb-89ff-6bb05f595be8.png">

Testing

1. Build, run and upload sample data into DDF
2. Verify autocomplete functionality
3. Verify search functionality with boolean statements
4. Verify that the boolean text carries over to the advanced search
5. Verify search functionality with the advanced search form
5. Verify boolean text carries over from advanced to basic
6. Add another search attribute (for example date.created), create invalid boolean text and verify that proper results are returned
